bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,483 @@
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, smartBatteryControl !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("smartBatteryControlText")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556484", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, smartBatteryControlText !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryChargeTimeFrom")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556239", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, batteryChargeTimeFrom !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("batteryChargeTimeTo")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556240", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, batteryChargeTimeTo !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("maxDepthOfDischarge")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556247", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug(
+                        "Unexpected error in KostalConfigurationHandler.configurationHandler, maxDepthOfDischarge !",
+                        e);
+            }
+        }
+
+        if (channelUID.getId().equals("shadowManagement")) {
+            valueConfiguration = command.toString();
+
+            try {
+                SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                        "33556483", valueConfiguration);
+            } catch (Exception e) {
+                logger.debug("Unexpected error in KostalConfigurationHandler.configurationHandler, shadowManagement !",
+                        e);
+            }
```

Comment:
You can save a lot of code if you refactor this into a utility method in this class.